lib_eio/process.mli

```diff
@@ -0,0 +1,148 @@
+    method virtual pid : int
+    method virtual await : exit_status
+    method virtual signal : int -> unit
+end
+
+val pid : #t -> int
+(** [pid t] is the process ID of [t]. *)
+
+val await : #t -> exit_status
+(** [await t] waits for subprocess [t] to exit and then reports the status. *)
+
+val await_exn : #t -> unit
+(** Like {! await} except an exception is raised if the status is not [`Exited 0]. *)
+
+val signal : #t -> int -> unit
+(** [signal t i] sends the signal [i] to process [t].
+
+    If the process has already exited then this does nothing
+    (it will not signal a different process, even if the PID has been reused).
+
+    See {!Sys} for the signal numbers. *)
+
+(** {2 Spawning processes} *)
+
+class virtual mgr : object
+  method virtual pipe :
+    sw:Switch.t ->
+    <Flow.source; Flow.close> * <Flow.sink; Flow.close>
+
+  method virtual spawn :
+    sw:Switch.t ->
+    ?cwd:Fs.dir Path.t ->
+    ?stdin:Flow.source ->
+    ?stdout:Flow.sink ->
+    ?stderr:Flow.sink ->
+    ?env:string array ->
+    ?executable:string ->
+    string list ->
+    t
+end
+(** A process manager capable of spawning new processes. *)
+
+val spawn :
+  sw:Switch.t ->
+  #mgr ->
+  ?cwd:#Fs.dir Path.t ->
+  ?stdin:#Flow.source ->
+  ?stdout:#Flow.sink ->
+  ?stderr:#Flow.sink ->
+  ?env:string array ->
+  ?executable:string ->
+  string list -> t
+(** [spawn ~sw mgr args] creates a new subprocess that is connected to the switch [sw].
+
+    The subprocess will be sent {! Sys.sigkill} when the switch is released.
+
+    If the flows [stdin], [stdout] and [stderr] are not backed by file descriptors then
+    this also creates pipes and spawns fibers to copy the data as necessary.
+    If you need more control over file descriptors, see {!Eio_unix.Process}.
+
+    @param cwd The current working directory of the process (default: same as parent process).
+    @param stdin The flow to attach to the process's standard input (default: same as parent process).
+    @param stdout A flow that the child process's standard output goes to (default: same as parent process).
+    @param stderr A flow that the child process's standard error goes to (default: same as parent process).
+    @param env The environment for the child (default: same as parent process).
+    @param executable The path of the executable to run.
+                      If not given then the first item in [args] is used,
+                      searching $PATH for it if necessary. *)
+
+val run :
+  #mgr ->
+  ?cwd:#Fs.dir Path.t ->
+  ?stdin:#Flow.source ->
+  ?stdout:#Flow.sink ->
+  ?stderr:#Flow.sink ->
+  ?env:string array ->
+  ?executable:string ->
+  string list -> unit
+(** [run] does {!spawn} followed by {!await_exn}, with the advantage that if the process fails then
+    the error message includes the command that failed.
+
+    Note: If [spawn] needed to create extra fibers to copy [stdin], etc, then it also waits for those to finish. *)
+
+val parse_out :
```

Comment:
Do you think we should also provide another convenience function that is `parse_out` with `Eio.Buf_read.take_all`, so a `pread` of sorts in [obuilder](https://github.com/ocurrent/obuilder/blob/master/lib/os.ml) speak I think. I just have a feeling most people will want that and it might not be obvious they can achieve that with those two.